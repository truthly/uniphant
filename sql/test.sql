BEGIN;

--
-- This only tests the extension can be installed
-- and calls a few functions that can easily be tested.
--
-- For a more complete integration test,
-- see .github/workflows/build-test.yml
--

CREATE EXTENSION uniphant CASCADE;

SELECT set_config('request.header.origin','http://localhost',FALSE);
SELECT effective_domain();

SELECT set_config('request.header.origin','http://example.com',FALSE);
SELECT effective_domain();

SELECT set_config('request.header.X-Forwarded-For','127.0.0.1',FALSE);
SELECT remote_ip();

SELECT set_config('request.header.X-Forwarded-For','192.168.123.123',FALSE);
SELECT remote_ip();

SELECT api.sign_up(username := 'test');
SELECT user_id, username FROM users;
SELECT user_id FROM access_tokens;
SELECT api.sign_out();

ROLLBACK;
