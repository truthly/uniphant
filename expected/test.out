BEGIN;
--
-- This only tests the extension can be installed
-- and calls a few functions that can easily be tested.
--
-- For a more complete integration test,
-- see .github/workflows/build-test.yml
--
CREATE EXTENSION uniphant CASCADE;
NOTICE:  installing required extension "webauthn"
NOTICE:  installing required extension "pguecc"
NOTICE:  installing required extension "pgcrypto"
NOTICE:  installing required extension "cbor"
SELECT set_config('request.header.origin','http://localhost',FALSE);
    set_config    
------------------
 http://localhost
(1 row)

SELECT effective_domain();
 effective_domain 
------------------
 localhost
(1 row)

SELECT set_config('request.header.origin','http://example.com',FALSE);
     set_config     
--------------------
 http://example.com
(1 row)

SELECT effective_domain();
 effective_domain 
------------------
 example.com
(1 row)

SELECT set_config('request.header.X-Forwarded-For','127.0.0.1',FALSE);
 set_config 
------------
 127.0.0.1
(1 row)

SELECT remote_ip();
 remote_ip 
-----------
 127.0.0.1
(1 row)

SELECT set_config('request.header.X-Forwarded-For','192.168.123.123',FALSE);
   set_config    
-----------------
 192.168.123.123
(1 row)

SELECT remote_ip();
    remote_ip    
-----------------
 192.168.123.123
(1 row)

SELECT api.sign_up(username := 'test');
 sign_up 
---------
 t
(1 row)

SELECT user_id, username FROM users;
 user_id | username 
---------+----------
       1 | test
(1 row)

SELECT user_id FROM access_tokens;
 user_id 
---------
       1
(1 row)

SELECT api.sign_out();
 sign_out 
----------
 t
(1 row)

ROLLBACK;
